project_report/tex/implementation.tex

```diff
@@ -86,3 +86,170 @@ \subsection{OPC-UA-Client}
+	\label{figure:ef_pilsner}
+\end{figure}
+
+The estimated error function seen in Figure \ref{figure:ef_pilsner} gives an
+idea of when the beer production machine starts to produce many rejected
+products. All the estimated error functions for the products the machine can
+produce can be seen in Table \ref{table:eef} and the graphs can be found in
+Appendix \ref{app:error_function_graphs}.
+
+\begin{table}[ht]
+     \begin{tabularx}{\textwidth}{|>{\RaggedRight}p{3cm}|>{\RaggedRight}X|>{\RaggedRight}p{3cm}|}
+     \hline
+     \textbf{Product type} & \textbf{Estimated Error Function} & \textbf{Accuracy} \\
+     \hline
+     Pilsner & \(y = 0,0003x^3 - 0,0291x^2 + 0,95x - 7,3838\) & \(R^2 = 0,9905\) \\
+     \hline
+     Wheat & \(y = 5E-05x^3 - 0,0077x^2 + 1,3163x - 3,6317\) & \(R^2 = 0,9983\) \\
+     \hline
+     IPA & \(y = -0,0001x^3 + 0,0548x^2 - 3,7529x + 72\) & \(R^2 = 0,9981\) \\
+     \hline
+     Stout & \(y = -0,0002x^3 + 0,0319x^2 - 1,551x + 67,071\) & \(R^2 = 0,9331\) \\
+     \hline
+     Ale & \(y = -2E-05x^3 + 0,0099x^2 - 0,5615x + 9,1857\) & \(R^2 = 1\) \\
+     \hline
+     Alcohol Free & \(y = 3E-05x^3 - 0,0021x^2 + 0,4464x + 10\) & \(R^2 = 0,9935\) \\
+     \hline
+    \end{tabularx}
+    \caption{Estimated Error Functions}
+    \label{table:eef}
+\end{table}
+
+\subsection{Optimal Production Speed}
+To find the optimal production speed the same data as when estimating the error
+function is used. As the error simulation in the machine is a function of the
+normalised machine speed and the selected product, the normalised machine speed
+must be calculated. This is done by taking 100 over the maximum machine speed.
+This multiplied by chosen machine speed will then give the percentage of the 
+maximum machine speed, which is used when plotting the graph for the optimal
+production speed. The normalised machine speed will be the \(x\)-values and the
+calculated OEE will be the \(y\)-values. An example can be seen in Figure
+\ref{figure:ops_pilsner}, where the graph for the optimal production speed for
+the product type pilsner can be seen.
+
+\begin{figure}[ht]
+	\centering
+	\includegraphics[width=1\linewidth]{images/ops/pilsner.png}
+	\caption{Optimal Production Speed for Pilsner}
+	\label{figure:ops_pilsner}
+\end{figure}
+
+This plot is generated using Microsoft Excel. When the OEE for each tested
+machine speed is plotted, a third degree polynomial trend line is made and an
+expression for the function generated is given. For the pilsner the function is
+given by:
+
+\[y = -0,0003x^3+0,0334x^2-0,1057x+9,0202\]
+
+
+With an \(R^2\) of \(0,9905\) which is used to see how close to the data points
+the polynomial is (the closer to 1 the more accurate).\\
+
+To find the maximum of the function, the derivative is calculated:
+
+\[y' = -\frac{9x^2-668x+1057}{10000}\]
+
+To find the maximum this equation should be solved for \(x\) when \(y=0\). This
+gives \(72,60463\), which is percentage of the maximum machine speed which is
```

Comment:
This gives an x-value of 72,60463, which is the optimal machine speed for the pilsner product type, measured in percentage.